crates/swc_ecma_transforms_proposal/src/decorators/legacy/metadata.rs

```diff
@@ -469,7 +465,10 @@ fn serialize_type(class_name: Option<&Ident>, param: Option<&TsTypeAnn>) -> Expr
             }
         }
 
-        *undefined(DUMMY_SP)
+        match u {
+            Some(i) => i,
```

Comment:
I think this fixes an issue where `string | never` => `void 0` instead of `String`

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAKgCwJZgOYBECmoBOBDKEHALhgAUcQAHTHKATy1wKJgF4YAKASnYD4YAbwC+AWABQEmDGAAbPBAjxkaIVOkwAAohQZsRFjnXSoK1Jyp58AWwCMpaDl0wAPjDCYAbrV4j1woA